File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
class Solution {
    public int[] solution(int[] answers) {
        int[] answer = new int[0];
        
        int u1 = 0; int[] arr1 = {1,2,3,4,5};
        int u2 = 0; int[] arr2 = {2,1,2,3,2,4,2,5};
        int u3 = 0; int[] arr3 = {3,3,1,1,2,2,4,4,5,5};
        
        for(int i=0; i<answers.length; i++) {
            if(arr1[i%arr1.length] == answers[i]) u1++;
            if(arr2[i%arr2.length] == answers[i]) u2++;
            if(arr3[i%arr3.length] == answers[i]) u3++;
        }
        
             if(u1 > u2 && u1 > u3) {answer = new int[1]; answer[0] = 1;}
        else if(u2 > u1 && u2 > u3) {answer = new int[1]; answer[0] = 2;}
        else if(u3 > u1 && u3 > u2) {answer = new int[1]; answer[0] = 3;}
        else if(u1 > u3 && u1 == u2) {answer = new int[2]; answer[0] = 1; answer[1] = 2;}
        else if(u1 > u2 && u1 == u3) {answer = new int[2]; answer[0] = 1; answer[1] = 3;}
        else if(u2 > u1 && u2 == u3) {answer = new int[2]; answer[0] = 2; answer[1] = 3;}
        else if(u1 == u2 && u1 == u3) {answer = new int[3]; answer[0] = 1; answer[1] = 2; answer[2] = 3;}
        
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = [-1, 0, 0, 0]
    
    num1 = [1, 2, 3, 4, 5]
    num2 = [2, 1, 2, 3, 2, 4, 2, 5]
    num3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    for i in range(len(answers)):
        if num1[i % len(num1)] == answers[i]: answer[1] += 1
        if num2[i % len(num2)] == answers[i]: answer[2] += 1
        if num3[i % len(num3)] == answers[i]: answer[3] += 1
        
    result = []
    
    max_score = answer[0]
    for i in range(1, 4):
        if max_score < answer[i]:
            result.clear()
            result.append(i)
            max_score = answer[i]
        elif max_score == answer[i]:
            result.append(i)
    
    return result